html2docx/image.py

```diff
@@ -0,0 +1,91 @@
+import io
+import pathlib
+import time
+import urllib.error
+import urllib.request
+from typing import Dict, Optional
+
+from docx.image.exceptions import UnrecognizedImageError
+from docx.image.image import Image
+from docx.shared import Inches
+
+USABLE_HEIGHT = Inches(8.1)
+USABLE_WIDTH = Inches(5.8)
+
+MAX_IMAGE_SIZE = 10 * 1024 * 1024  # 10 MiB
+
+
+def load_image(src: str) -> io.BytesIO:
+    image_buffer = None
+    retry = 3
+    while retry and not image_buffer:
+        try:
+            with urllib.request.urlopen(src) as response:
+                size = response.getheader("Content-Length")
+                if size and int(size) > MAX_IMAGE_SIZE:
+                    break
+                data = response.read(MAX_IMAGE_SIZE + 1)
+        except (ValueError, urllib.error.HTTPError):
```

Comment:
Is this `ValueError` here to guard against non-int header? WDYT about adding a comment here as well?

Reply:
`ValueError` guards against malformed urls, removing it makes `test_bad_url` fail. It also guards against malformed responses, but that is untested. Perhaps it’ll be easier to test with the local http server solution. I’ll give it a shot.